BLUESPAWN-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -6,296 +6,296 @@
-{
-	return QueryEvents(channel, id, std::set<std::wstring>(), reaction);
-}
-
-int EventLogs::QueryEvents(const wchar_t* channel, unsigned int id, std::set<std::wstring>& params, Reaction& reaction)
-{
-	DWORD status = ERROR_SUCCESS;
-	EVT_HANDLE hResults = NULL;
-
-	auto query = std::wstring(L"Event/System[EventID=") + std::to_wstring(id) + std::wstring(L"]");
-	auto wquery = query.c_str();
-
-	hResults = EvtQuery(NULL, channel, wquery, EvtQueryChannelPath | EvtQueryReverseDirection);
-	if (NULL == hResults)
-	{
-		status = GetLastError();
-
-		if (ERROR_EVT_CHANNEL_NOT_FOUND == status)
-			LOG_ERROR("EventLogs::QueryEvents: The channel was not found.");
-		else if (ERROR_EVT_INVALID_QUERY == status)
-			// You can call the EvtGetExtendedStatus function to try to get 
-			// additional information as to what is wrong with the query.
-			LOG_ERROR(L"EventLogs::QueryEvents: The query " + query + L" is not valid.");
-		else
-			LOG_ERROR("EventLogs::QueryEvents: EvtQuery failed with " + std::to_string(status));
-
-		goto cleanup;
-	}
+	std::shared_ptr<EVENT_DETECTION> EventLogs::EventLogItemToDetection(const EventLogItem& pItem) {
+		auto detect = std::make_shared<EVENT_DETECTION>(0, 0, L"", L"", L"");
 
-	int numFound = 0;
-	status = ProcessResults(hResults, reaction, &numFound, params);
+		detect->eventID = pItem.GetEventID();
+		detect->channel = pItem.GetChannel();
+		detect->eventRecordID = pItem.GetEventRecordID();
+		detect->timeCreated = pItem.GetTimeCreated();
+		detect->rawXML = pItem.GetXML();
+		detect->params = pItem.GetProperties();
 
-cleanup:
+		return detect;
+	}
+
+	bool IsChannelOpen(const std::wstring& channel) {
+		EVT_HANDLE hChannel = NULL;
+		DWORD status = ERROR_SUCCESS;
+		PEVT_VARIANT pProperty = NULL;  
+		PEVT_VARIANT pTemp = NULL;
+		DWORD dwBufferSize = 0;
+		DWORD dwBufferUsed = 0;
+
+		// Open the channel config
+		hChannel = EvtOpenChannelConfig(NULL, channel.c_str(), 0);
+		if (NULL == hChannel)
+		{
+			LOG_ERROR(L"EventLogs::IsChannelOpen: EvtOpenChannelConfig failed with " + std::to_wstring(GetLastError()) + L" for channel " + channel);
+			return false;
+		}
 
-	if (hResults)
-		EvtClose(hResults);
+		// Attempt to get the channel property
+		if (!EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed))
+		{
+			status = GetLastError();
+			if (ERROR_INSUFFICIENT_BUFFER == status) {
+				dwBufferSize = dwBufferUsed;
+				pTemp = (PEVT_VARIANT)realloc(pProperty, dwBufferSize);
+
+				if (pTemp) {
+					pProperty = pTemp;
+					pTemp = NULL;
+					EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed);
+				}
+				else {
+					if (pProperty)
+						free(pProperty);
+
+					LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+					return false;
+				}
+			}
 
-	if (status == ERROR_SUCCESS)
-		return numFound;
-	return -1;
-}
+			if (ERROR_SUCCESS != (status = GetLastError())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5092780)